src/Authenticator/HttpBasicAuthenticator.php

```diff
@@ -91,7 +91,7 @@ public function unauthorizedChallenge(ServerRequestInterface $request)
     protected function loginHeaders(ServerRequestInterface $request)
     {
         $server = $request->getServerParams();
-        $realm = $this->getConfig('realm') ?: $server['SERVER_NAME'];
+        $realm = $this->getConfig('realm') === null ? $server['SERVER_NAME'] : $this->getConfig('realm');
```

Comment:
This is a case where weak totally suffices and is usually even more correct.

Reply:
So '0' is not a valid realm? Who says that? I will add a test against trimmed ''-string?

Reply:
Just wondering about people and their configs. Could also come from DB and accidently be an empty string. Either way shouldnt run into it, no need to undo the current weak check and break stuff for no reason IMO.

Reply:
I don't share that philosophy obviously. Weak types create a ton of problems for me and PHP 7.1/7.2 will be harder to upgrade to if we allow implicit weak "forgiving" types everywhere.
I can add a trim() === '' check however easily.

Reply:
Never mind then.

Reply:
`if(!strlen($string))` works fine for rejecting `''|null`

Reply:
Or just keeping the `?:`.